modules/standard/IO.chpl

```diff
@@ -3318,51 +3317,48 @@ inline proc fileWriter.unlock() {
   }
 }
 
+@chpldoc.nodoc
+private inline proc offsetHelper(fileRW) {
+  var ret:int(64);
+  on fileRW._home {
+    if !fileRW.locking then try! fileRW.lock();
+    ret = qio_channel_offset_unlocked(fileRW._channel_internal);
+    if !fileRW.locking then fileRW.unlock();
```

Comment:
Okay, that's a good point! Then I don't think we can directly change the locking behavior of `offset` without somehow warning users. One way of doing this, is using a config-param to toggle between the new and deprecated behaviors.

I'm not sure if you've set something like this up yet, so as an example: `file.path` used to return a relative path, but was updated to only return absolute paths. We didn't want to remove the version that returns relative paths without warning users, so there is a config param called `filePathAbsolute` in the IO module that toggles whether the deprecated or new overload of `file.path` is called. It defaults to the deprecated version and triggers a warning saying that the parameter should be switched to use the new behavior.

You could do something similar here. I.e., have a locking and non-locking version of `offset` and a parameter to control which overload is called (using where-clauses). It would default to the locking version to avoid introducing races into their code.

Reply:
I have updated the PR with this change, the flag is called `fileOffsetWithoutLocking` (definitely open to comments on this name). I also added a `chpl_offset` which allows us to get the new behavior with setting the flag, this allows me to change all of the `_offset`'s in the standard library to the new behavior. `chpl_offset` is for standard modules to avoid warnings about `fileOffsetWithoutLocking` propagating everywhere, it should be removed when `fileOffsetWithoutLocking` is.

Reply:
I think I will keep it as is, as it matches the casing for `filePathAbsolute`.

Also, expect more changes as I update all the modules/tests which use `offset` :smile: